tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Host() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->host,
+				$config->searchengine->solr->default->service->default->endpoint->primary->host
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Port() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->port,
+				$config->searchengine->solr->default->service->default->endpoint->primary->port
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Path() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->path,
+				$config->searchengine->solr->default->service->default->endpoint->primary->path
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testMissingConfigParamLogPrepareXml() {
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset( $config->log->prepare->xml );
+
+			return $config;
+		} );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testEmptyConfiguration() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array( 'solr' => array( 'default' => array( 'service' => array( 'index' => array( 'endpoint' => array( 'primary' => array( 'path' => '' ) ) ) ) ) ) )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testInvalidConfiguration() {
```

Comment:
Basically this test doesn't make sense to me anymore as it isn't _unit testing_ some code of Opus4. Now it might be testing if Opus4 is properly set up as an application which isn't intention of unit tests. Thus it might be required to move this part to some selenium tests checking whether some vanilla installation of Opus4 is capable to perform certain actions or not.

Aside from that, Opus_Search_Config is a separate class basically providing opportunity to check the way of handling configuration locally. Solarium in turn is best place for checking how Solarium is handling invalid configuration. Though I understand your concerns it isn't part of _unit testing_ Opus4 code base anymore for it isn't actually processing this configuration in any particular way.